sim/shaman/items.go

```diff
@@ -176,20 +216,68 @@ var ItemSetSpiritwalkersRegalia = core.NewItemSet(core.ItemSet{
 					shaman.AddStatDynamic(sim, stats.HasteRating, float64(changedHasteRating))
 				},
 			})
-			shaman.RegisterAura(core.Aura{
-				Label:    "Spiritwalker's Regalia 4P",
-				Duration: core.NeverExpires,
-				OnReset: func(aura *core.Aura, sim *core.Simulation) {
+
+			setBonusAura.AttachProcTrigger(core.ProcTrigger{
+				Name:           "Spiritwalker's Regalia 4P",
+				ClassSpellMask: SpellMaskOverload,
+				Callback:       core.CallbackOnCastComplete,
+				Handler: func(sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
+					procAura.Activate(sim)
+					procAura.AddStack(sim)
+				},
+			})
+		},
+	},
+})
+
+// T13 Resto
+// (2) Set: After using Mana Tide Totem, the cost of your healing spells are reduced by 25% for 15 sec.
+// (4) Set: Increases the duration of Spiritwalker's Grace by 5 sec, and you gain 30% haste while Spiritwalker's grace is active.
+var ItemSetSpiritwalkersVestments = core.NewItemSet(core.ItemSet{
+	Name: "Spiritwalker's Vestments",
+	Bonuses: map[int32]core.ApplySetBonus{
+		2: func(_ core.Agent, _ *core.Aura) {
+			// Not implemented
+		},
+		4: func(agent core.Agent, setBonusAura *core.Aura) {
+			shaman := agent.(ShamanAgent).GetShaman()
+
+			hasteMulti := 1.30
+			aura := shaman.RegisterAura(core.Aura{
+				Label:    "Item - Shaman T13 Restoration 4P Bonus (Spiritwalker's Grace)",
+				ActionID: core.ActionID{SpellID: 105876},
+				Duration: shaman.spiritwalkersGraceBaseDuration() + 5*time.Second,
+				OnGain: func(aura *core.Aura, sim *core.Simulation) {
+					shaman.MultiplyCastSpeed(hasteMulti)
+					shaman.MultiplyAttackSpeed(sim, hasteMulti)
+				},
+				OnExpire: func(aura *core.Aura, sim *core.Simulation) {
+					shaman.MultiplyCastSpeed(1 / hasteMulti)
+					shaman.MultiplyAttackSpeed(sim, 1/hasteMulti)
+				},
+			})
+
+			setBonusAura.AttachProcTrigger(core.ProcTrigger{
+				Name:           "Item - Shaman T13 Restoration 4P Bonus (Spiritwalker's Grace) - Proc",
+				ActionID:       core.ActionID{SpellID: 105876},
+				ClassSpellMask: SpellMaskSpiritwalkersGrace,
+				Callback:       core.CallbackOnApplyEffects,
+				Handler: func(sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
 					aura.Activate(sim)
 				},
-				//TODO does dtr overloads proc this ?
-				OnCastComplete: func(aura *core.Aura, sim *core.Simulation, spell *core.Spell) {
-					if spell.ClassSpellMask&SpellMaskOverload > 0 {
-						procAura.Activate(sim)
-						procAura.AddStack(sim)
-					}
+			})
+
+			setBonusAura.AttachSpellMod(core.SpellModConfig{
+				Kind:      core.SpellMod_Custom,
+				ClassMask: SpellMaskSpiritwalkersGrace,
+				ApplyCustom: func(mod *core.SpellMod, spell *core.Spell) {
+					shaman.SpiritwalkersGraceAura.Duration = shaman.spiritwalkersGraceBaseDuration() + 5*time.Second
+				},
+				RemoveCustom: func(mod *core.SpellMod, spell *core.Spell) {
+					shaman.SpiritwalkersGraceAura.Duration = shaman.spiritwalkersGraceBaseDuration()
```

Comment:
Can you not just do a += and -= 5 seconds here rather than needing to set the values directly?

Reply:
I think I can, I will try. This is just a remnant of the old implementation copy pasted to the new SpellMod